src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,30 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod)), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * static cache around $this->contextFactory->createFromReflector()
+     * prevents a lot of redundant calls to createForNamespace.
+     */
+    private function createFromReflector(\Reflector $reflector): Context
+    {
+        if (method_exists($reflector, 'getDeclaringClass')) {
+            $reflector = $reflector->getDeclaringClass();
+        }
+
+        $cacheKey = $reflector->getNamespaceName().':'.$reflector->getFileName();
```

Comment:
I don't get the cache key: the namespace looks embedded in the filename, so we could remove it, right?
then for methods, all methods have the same cache key? is this legit? if yes, why not use the class name instead? if not, that's buggy, right?